crates/net/network/src/state.rs

```diff
@@ -443,46 +437,51 @@ impl<N: NetworkPrimitives> NetworkState<N> {
                 }
             }
 
-            // need to buffer results here to make borrow checker happy
-            let mut closed_sessions = Vec::new();
-            let mut received_responses = Vec::new();
-
-            // poll all connected peers for responses
-            for (id, peer) in &mut self.active_peers {
-                if let Some(mut response) = peer.pending_response.take() {
-                    match response.poll(cx) {
-                        Poll::Ready(res) => {
-                            // check if the error is due to a closed channel to the session
-                            if res.err().is_some_and(|err| err.is_channel_closed()) {
-                                debug!(
-                                    target: "net",
-                                    ?id,
-                                    "Request canceled, response channel from session closed."
-                                );
-                                // if the channel is closed, this means the peer session is also
-                                // closed, in which case we can invoke the [Self::on_closed_session]
-                                // immediately, preventing followup requests and propagate the
-                                // connection dropped error
-                                closed_sessions.push(*id);
-                            } else {
-                                received_responses.push((*id, res));
+            loop {
+                // need to buffer results here to make borrow checker happy
+                let mut closed_sessions = Vec::new();
+                let mut received_responses = Vec::new();
+
+                // poll all connected peers for responses
+                for (id, peer) in &mut self.active_peers {
+                    if let Some(mut response) = peer.pending_response.take() {
```

Comment:
this predates `let Some() {else }` 

looks like we can use `let Some(mut response) = .. {else continue}` here and remove one ident?